gateway/handler_tar.go

```diff
@@ -16,23 +15,25 @@ import (
 
 var unixEpochTime = time.Unix(0, 0)
 
-func (i *handler) serveTAR(ctx context.Context, w http.ResponseWriter, r *http.Request, resolvedPath ipath.Resolved, contentPath ipath.Path, begin time.Time, logger *zap.SugaredLogger) bool {
-	ctx, span := spanTrace(ctx, "ServeTAR", trace.WithAttributes(attribute.String("path", resolvedPath.String())))
+func (i *handler) serveTAR(ctx context.Context, w http.ResponseWriter, r *http.Request, imPath ImmutablePath, contentPath ipath.Path, begin time.Time, logger *zap.SugaredLogger) bool {
+	ctx, span := spanTrace(ctx, "ServeTAR", trace.WithAttributes(attribute.String("path", imPath.String())))
 	defer span.End()
 
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	// Get Unixfs file
-	file, err := i.api.GetUnixFsNode(ctx, resolvedPath)
-	if err != nil {
-		err = fmt.Errorf("error getting UnixFS node for %s: %w", html.EscapeString(contentPath.String()), err)
-		webError(w, err, http.StatusInternalServerError)
+	// Get Unixfs file (or directory)
+	gwMetadata, file, err := i.api.GetAll(ctx, imPath)
```

Comment:
nit; we've already renamed type to `ContentPathMetadata`, would be good to do the same with all vars for readability
```suggestion
	pathMetadata, file, err := i.api.GetAll(ctx, imPath)
```